packages/matter-node.js/src/matter/MatterDevice.ts

```diff
@@ -60,8 +69,61 @@ export class MatterDevice {
         return this;
     }
 
-    start() {
-        this.broadcasters.forEach(broadcaster => broadcaster.announce());
+    async start() {
+        /*for (const broadcaster of this.broadcasters) {
+            broadcaster.setCommissionMode(0, this.deviceName, this.deviceType, this.vendorId, this.productId, this.discriminator);
+            broadcaster.announce();
+        }*/
+        this.startAnnouncement();
+    }
+
+    startAnnouncement() {
+        this.announcementStartedTime = Time.nowMs();
+        this.announceInterval = Time.getPeriodicTimer(DEVICE_ANNOUNCEMENT_INTERVAL, () => this.announce()).start();
+        this.announce();
+    }
+
+    announce(announceOnce = false) {
```

Comment:
This PR is mixing three completely different things:
1) configurable port
2) configurable filtering of network interface
3) add announce logic to device

In the future, please separate disparate functionality like this into separate PRs.  It make approving the small stuff (like configurable port) a lot easier.

Reply:
Yes sorry for that